src/lazy/binary/raw/v1_1/immutable_buffer.rs

```diff
@@ -1012,6 +1007,9 @@ impl<'a> BinaryBuffer<'a> {
         let args_length = args_length_flex_uint.value() as usize;
 
         let total_length = header_length + args_length;
+        if self.len() < total_length {
+            return IonResult::incomplete("a length-prefixed e-expression", self.offset);
+        }
```

Comment:
🪧 This error happened to be surfaced by the test data I was using, so I went ahead and patched it while I was in here.